### PR TITLE
owner: add recover panic in owner campaignLoop.

### PR DESCRIPTION
**Pull request by winkyao**

### CODE REVIEW

File: owner/manager.go [hunk 1]
```diff
@@ -172,7 +173,23 @@ func (m *ownerManager) CampaignOwner(ctx context.Context) error {
 	return nil
 }
 
+func recoverInOwner(funcName string, quit bool) {
+	r := recover()
```
Comment (ngaut) [inline owner/manager.go]: Please use if assignment pattern.

**Comment by winkyao** [comment]
@ngaut PTAL

File: owner/manager.go [hunk 2]
```diff
@@ -172,7 +173,21 @@ func (m *ownerManager) CampaignOwner(ctx context.Context) error {
...
+		if quit {
+			// Wait for metrics to be pushed.
+			time.Sleep(time.Second * 15)
+			os.Exit(1)
+		}
+	}
+}
+
 func (m *ownerManager) campaignLoop(ctx context.Context, etcdSession *concurrency.Session) {
+	defer recoverInOwner("campaignLoop", true)
```
Comment (coocood) [inline owner/manager.go]: We don't need to quit if ddl doesn't work.

File: owner/manager.go [hunk 3]
```diff
@@ -172,7 +173,21 @@ func (m *ownerManager) CampaignOwner(ctx context.Context) error {
 	return nil
 }
 
+func recoverInOwner(funcName string, quit bool) {
```
Comment (ngaut) [inline owner/manager.go]: The function acts like graceful exit process, the function name is not consistent with the behavior.

**Comment by winkyao** [comment]
@coocood PTAL

File: owner/manager.go [hunk 4]
```diff
@@ -172,7 +173,21 @@ func (m *ownerManager) CampaignOwner(ctx context.Context) error {
...
+	if r := recover(); r != nil {
+		buf := util.GetStack()
+		log.Errorf("%s, %v, %s", funcName, r, buf)
+		metrics.PanicCounter.WithLabelValues(metrics.LabelDDLOwner).Inc()
+		if quit {
```
Comment (ngaut) [inline owner/manager.go]: How about remove the quit code block which would make it a real recover function?

**Comment by winkyao** [comment]
@ngaut PTAL

**Comment by ngaut** [comment]
LGTM

**Pull request merged**